packages/react-menu/src/components/MenuTrigger/useTriggerElement.ts

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import { useMergedRefs } from '@fluentui/react-utilities';
+import { useKeyboardNavigationState } from '@fluentui/react-focus-management';
+import { MenuTriggerState } from './MenuTrigger.types';
+import { useMenuContext } from '../../menuContext';
+
+// Helper type to select on parts of state the hook uses
+type UseTriggerElementState = Pick<MenuTriggerState, 'children'>;
+
+export const useTriggerElement = (state: UseTriggerElementState): MenuTriggerState => {
+  const triggerRef = useMenuContext(context => context.triggerRef);
+  const setOpen = useMenuContext(context => context.setOpen);
+  const open = useMenuContext(context => context.open);
+  const triggerId = useMenuContext(context => context.triggerId);
+  const on = useMenuContext(context => context.on);
+  const { isNavigatingWithKeyboard } = useKeyboardNavigationState();
+
+  const child = React.Children.only(state.children);
```

Comment:
Do you want to add a TODO to add an additional warning when `child.type` is a `React.Fragment`?

Reply:
Yeah, AFAIK the check isn't exactly a one liner and `react-is` does the most thorough one